controller.py

```diff
@@ -1,47 +1,114 @@
 from view import View
 from readchar import readkey, key
-from model import load_shortlist,load_applicants
+from model import load_shortlist
 
 class Controller:
 
     def __init__(self,path):
         self.path = path
         self.shortlist = load_shortlist(path)
+        self.current_view = "home"
         self.view = View()
 
     def show_boot_message(self):
         self.view.boot_message(self.path,len(self.shortlist.applicants))
-
-    def show_applicant_info(self):
-        self.view.view_applicants(self.shortlist.applicants)
+        self.current_view = "home"
 
     def show_criteria(self):
         self.view.view_criteria(self.shortlist.role,self.shortlist.role.criteria)
-    
+        self.current_view = "criteria"
+
     def show_role_info(self):
         self.view.view_role(self.shortlist.role)
+        self.current_view = "role"
 
-    def show_shortlist(self):
-        self.view.view_shortlist(self.shortlist)
+    def show_applicants_list(self):
+        self.view.view_applicants_list(self.shortlist)
+        self.current_view = "applicants_list"
+
+    def show_applicant_details(self):
+        try:
+            i = int(input("Please enter the applicant number:"))
+            self.view.view_applicant_details(self.shortlist.applicants[i-1])
+            self.current_view = "applicant_details"
+        except (ValueError, IndexError):
+            pass
+    
 
     def run(self):
+
         self.show_boot_message()
 
         while True:
-            options = {"a":self.show_applicants,
-                       "b":self.show_boot_message,
-                       "c":self.show_criteria,
-                       "r":self.show_role_info,
-                       "s":self.show_shortlist,} 
-
             k = readkey()
-            output = options.get(k)
-            if output is not None:
-                output()
-            if k == "o":
-                print("(´⊙ω⊙`)!")
-            if k == key.CTRL_U:
-                print("ctrl u")
-            if k == key.ESC:
-                print("exiting the program...")
-                break
+
+            if self.current_view == "home":
+                options = {"r":self.show_role_info,
+                           "a":self.show_applicants_list}
+                
+                output = options.get(k)
+                if output is not None:
+                    output()
+                if k == key.ESC:
+                    print("exiting the program...")
+                    break
```

Comment:
As hitting the escape key always exits the program, you can move this logic to the top, outside of the conditionals on views. Then remove the escape key handling from all the view-specific blocks.